pkg/commands/add.go

```diff
@@ -66,7 +71,7 @@ func (a *AddCommand) ExecuteCommand(config *v1.Config, buildArgs *dockerfile.Bui
 				return err
 			}
 			logrus.Infof("Adding remote URL %s to %s", src, urlDest)
-			if err := util.DownloadFileToDest(src, urlDest); err != nil {
+			if err := util.DownloadFileToDest(src, urlDest, uid, gid); err != nil {
```

Comment:
The only place conversions are needed seems to be here then. `int64(uid), int64(gid)`

Reply:
I'm sorry, I'm not sure I'm following. Most methods in `fs_util.go` are expecting a int64 u/gid, with the exception of CreateFile (which is a bit weird IMO). Do you suggest to remove the casting from `util.GetUserGroup()` and perform it here?  I am not sure what'd be the advantage of that.

Reply:
two casts instead of four would be introduced.

Reply:
I did not add the method `GetUserGroup()`, it was already there - I just moved it to the util file.
If it would return uint32 instead of int64, then in `copy.go` you'd need to perform casting twice, when passing these methods to util.CopyDir() & util.CopyFile().

Reply:
I didn't take the time to read this in a bigger context. Good job!